.travis.yml

```diff
@@ -16,13 +16,27 @@ matrix:
       script: ci/run.sh
     - install: true
       script: ci/dox.sh
+    - env: RUSTFMT=On TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+  allow_failures:
+    - env: RUSTFMT=On TARGET=x86_64-unknown-linux-gnu NO_ADD=1
 
 install:
-  - if [ "$NO_ADD" = "" ]; then rustup target add $TARGET; fi
+  - if [ "$NO_ADD" == "" ]; then rustup target add $TARGET; fi
 
 script:
   - cargo generate-lockfile
-  - ci/run-docker.sh $TARGET
+  - if [[ "${RUSTFMT}" == "" ]]; then ci/run-docker.sh $TARGET; fi
```

Comment:
I think instead of this conditional you should be able to just specify `script:` above, right? (in the `include:` matrix)

Reply:
What do you mean? I added the conditional to avoid running the docker tests in the build that just checks formatting. I don't like this much either (in particular if we add clippy on top). 

Maybe a better way could be to move this to a `ci/run.sh` script that reads the environment variables (e.g. `RUSTFMT`, `CLIPPY`, etc) and does the right thing.

Reply:
Anyways since basically every PR breaks this one, and updating it sometimes requires manual edits, I'd rather merge it first, and improve the build scripts later.

Reply:
I believe you can avoid this if/else statement if you add a `script:` directive to the `env:` that you added in this same file

Reply:
I think i've fixed this here and in the enable_clippy PR.